javaslang/src/main/java/javaslang/collection/IndexedSeq.java

```diff
@@ -220,19 +221,30 @@ default int lastIndexWhere(Predicate<? super T> predicate, int end) {
 
     @Override
     default Iterator<T> reverseIterator() {
-        return new AbstractIterator<T>() {
-            private int i = IndexedSeq.this.length();
+        return new AbstractIterator<T>() { /*@formatter:off*/
+            final ListIterator<T> delegate = IndexedSeq.this.listIterator(IndexedSeq.this.length());
+            @Override public boolean hasNext() { return delegate.hasPrevious(); }
+            @Override public T getNext()       { return delegate.previous(); }
+        }; /*@formatter:on*/
+    }
 
-            @Override
-            public boolean hasNext() {
-                return i > 0;
-            }
+    default ListIterator<T> listIterator(int index) {
```

Comment:
@danieldietrich, this formatting makes it very easy to see that the logic is extremely simple: we're basically just delegating.
If everything would be on separate lines, we would lose this compactness.
If you don't agree, I will reformat it :)